File: sdk/rust/sage-client/examples/basic_usage.rs
//! Basic usage example for SAGE Rust client

use sage_client::{Client, ClientConfig};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize client
    let config = ClientConfig::new("http://localhost:8080");
    let mut client = Client::new(config).await?;

    println!("SAGE Rust Client - Basic Usage Example");
    println!("======================================\n");

    // Register agent
    let client_did = "did:sage:ethereum:0xAlice";
    client.register_agent(client_did, "Alice Agent").await?;
    println!("✓ Registered: {}", client_did);

    // Check server health
    let health = client.health_check().await?;
    println!("✓ Server status: {}", health.status);

    // Get server DID
    let server_did = client.get_server_did().await?;
    println!("✓ Server DID: {}", server_did);

    // Initiate handshake
    let session_id = client.handshake(&server_did).await?;
    println!("✓ Session established: {}", session_id);

    // Send message
    let message = b"Hello, Server!";
    let response = client.send_message(&session_id, message).await?;
    println!(
        "✓ Response: {}",
        String::from_utf8_lossy(&response)
    );

    // Send another message
    let message2 = b"How are you?";
    let response2 = client.send_message(&session_id, message2).await?;
    println!(
        "✓ Response 2: {}",
        String::from_utf8_lossy(&response2)
    );

    // Show active sessions
    println!("\n✓ Active sessions: {}", client.active_sessions());

    println!("\n✓ All operations completed successfully!");

    Ok(())
}

File: sdk/java/sage-client/examples/BasicUsage.java
import com.sage.client.*;
import com.sage.client.types.HealthStatus;

/**
 * Basic usage example for SAGE Java client
 */
public class BasicUsage {
    public static void main(String[] args) {
        try {
            System.out.println("SAGE Java Client - Basic Usage Example");
            System.out.println("======================================\n");

            // Initialize client
            ClientConfig config = ClientConfig.builder("http://localhost:8080")
                    .timeoutSeconds(30)
                    .maxSessions(100)
                    .build();

            SageClient client = new SageClient(config);

            // Register agent
            String clientDid = "did:sage:ethereum:0xAlice";
            client.registerAgent(clientDid, "Alice Agent");
            System.out.println("✓ Registered: " + clientDid);

            // Check server health
            HealthStatus health = client.healthCheck();
            System.out.println("✓ Server status: " + health.getStatus());

            // Get server DID
            String serverDid = client.getServerDid();
            System.out.println("✓ Server DID: " + serverDid);

            // Initiate handshake
            String sessionId = client.handshake(serverDid);
            System.out.println("✓ Session established: " + sessionId);

            // Send message
            byte[] message = "Hello, Server!".getBytes();
            byte[] response = client.sendMessage(sessionId, message);
            System.out.println("✓ Response: " + new String(response));

            // Send another message
            byte[] message2 = "How are you?".getBytes();
            byte[] response2 = client.sendMessage(sessionId, message2);
            System.out.println("✓ Response 2: " + new String(response2));

            // Show active sessions
            System.out.println("\n✓ Active sessions: " + client.activeSessions());

            System.out.println("\n✓ All operations completed successfully!");

        } catch (SageException e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
